### PR TITLE
chore(build): remove unused packagese, add needed packages, add eslint caching

### PR DESCRIPTION
**Pull request by jmetev1**
Checklist:

- [x ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ x] My pull request targets the `main` branch of freeCodeCamp.
- [x ] I have tested these changes either locally on my machine, or GitPod.

Closes none 

I wanted to make a smaller pr to remove dependencies for #47971 . Previously I had some trouble with linting so to do this I wanted to be able to lint faster. So I added caching to eslint. If that's not desired, I could take it out. Before this change, lint:js took a minute on my local. After, it takes 1 second. Also if you want, we could enable it on ci: https://enochchau.com/blog/2022/eslint-cache-ci/. I could add that here or as a separate pr.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -176,6 +174,7 @@
     "copy-webpack-plugin": "9.1.0",
     "core-js": "2.6.12",
     "dotenv": "16.0.3",
+    "eslint-plugin-testing-library": "^3.9.0",
```
Comment (jmetev1) [inline client/package.json:177]: added to avoid peer dependency warning

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
   };
 
   // Keep the tool panel visible when mobile address bar and/or keyboard are in view.
-  setToolPanelPosition = () => {
+  setToolPanelPosition = (): void => {
```
Comment (jmetev1) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:51]: some changes in typescript eslint packages started flagging this. which seems like a good thing right?

File: package.json [hunk 3]
```diff
@@ -114,7 +112,9 @@
     "@types/lodash": "4.14.194",
     "@types/node": "18.16.0",
     "@types/store": "2.0.2",
+    "@typescript-eslint/eslint-plugin": "^4.33.0",
```
Comment (jmetev1) [inline package.json:115]: this adds explicitly a dependency that we were using in eslintrc file.

File: package.json [hunk 4]
```diff
@@ -114,7 +112,9 @@
...
     "babel-jest": "29.5.0",
+    "babel-eslint": "^10.1.0",
```
Comment (jmetev1) [inline package.json:117]: eslintrc was using this already but we never explicitly installed it. Just happened to have it as a sub dependency.

**Comment by jmetev1** [comment]
all the tests passed last time except firefox tests. But i see on another branch they just ran them again and they passed so I'm now doing the same. I looked for a way to retrigger just that test but didn't see a way so i pulled from main and pushed.

File: package.json [hunk 5]
```diff
@@ -128,7 +128,6 @@
     "eslint-plugin-jest-dom": "3.9.4",
     "eslint-plugin-jsx-a11y": "6.7.1",
     "eslint-plugin-no-only-tests": "2.6.0",
-    "eslint-plugin-prefer-object-spread": "1.2.1",
```
Comment (jmetev1) [inline package.json:131]: not being used

File: client/package.json [hunk 6]
```diff
@@ -70,7 +70,6 @@
     "date-fns": "2.30.0",
     "enzyme": "3.11.0",
     "enzyme-adapter-react-16": "1.15.7",
-    "eslint": "7",
```
Comment (jmetev1) [inline client/package.json:73]: not used

File: client/package.json [hunk 7]
```diff
@@ -137,7 +136,6 @@
   },
   "devDependencies": {
     "@babel/plugin-syntax-dynamic-import": "7.8.3",
-    "@babel/types": "7.22.3",
```
Comment (jmetev1) [inline client/package.json:140]: not used

**Comment by jmetev1** [comment]
can my pr be reviewed even with pnpm lock conflicts? I've been assuming no but that mean I've been constantly fixing the lock file every time someone else gets merged.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks, @jmetev1. Local caching seems like a good idea, but I'd rather not do that in CI just to err on the side of caution.

 I just found the one other issue:

**Comment by ojeytonwilliams** [comment]
> can my pr be reviewed even with pnpm lock conflicts?

Absolutely. It just can't be merged until they're resolved.

File: package.json [hunk 8]
```diff
@@ -67,8 +67,7 @@
     "knip:all": "pnpm dlx knip@1",
     "prelint": "pnpm run -F=client predevelop",
     "lint": "npm-run-all create:* -p lint:*",
-    "lint:challenges": "cd ./curriculum && pnpm run lint",
-    "lint:js": "eslint --max-warnings 0 .",
+    "lint:js": "eslint --cache --cache-strategy content --max-warnings 0 .",
```
[the remark this answers is not in the document]
Comment (jmetev1) [reply package.json:72]: was this a reference to this line i took out? good eye. i caught it myself and fixed it, but didn't push! whoops

[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply package.json:72]: No, I misread the diff :man_facepalming: Sorry! I'll suggest a fix (a real one this time)

File: package.json [hunk 9]
```diff
@@ -68,7 +68,8 @@
     "prelint": "pnpm run -F=client predevelop",
     "lint": "npm-run-all create:* -p lint:*",
     "lint:challenges": "cd ./curriculum && pnpm run lint",
-    "lint:js": "eslint --max-warnings 0 .",
+    "lint:challenges": "cd ./curriculum && pnpm run lint",
+    "lint:js": "eslint --cache --cache-strategy content --max-warnings 0 .",
```
[the remark this answers is not in the document]
Comment (jmetev1) [reply package.json]: yeah i had added that part specifically for ci caching. i can take it out.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks for bearing with me @jmetev1 

This all LGTM :+1:

**Comment by jmetev1** [comment]
thanks!

**Review by moT01 — APPROVED** [review]
LGTM @jmetev1 🎉 Thanks for contributing to freeCodeCamp 🎉

**Pull request merged**